auth.js

```diff
@@ -0,0 +1,95 @@
+const config = require('./config')
+const debug = require('debug')('revshare-pointer:auth')
+const jwt = require('./jwt')
+
+const auth = {
+  authenticate: async function (ctx, next) {
+    if (!ctx.get('authorization')) {
+      ctx.throw(401, 'authentication failed. must provide authorization header')
+      return
+    }
+
+    const authHeader = ctx.get('authorization').split(' ')
+    const type = authHeader[0]
+    const token = authHeader[1]
+
+    if (type !== 'Bearer') {
+      ctx.throw(401, 'authentication failed. must provide authorization header with type, Bearer')
+      return
+    }
+
+    if (!token) {
+      ctx.throw(401, 'authentication failed. must provide authorization header with credentials')
+      return
+    }
+
+    debug('checking auth token. given=' + token,
+      'token=' + config.token,
+      'eq=' + (token === config.token))
+
+    if(token !== config.token) {
```

Comment:
checks like this are vulnerable to timing attacks: https://thisdata.com/blog/timing-attacks-against-string-comparison/

We should use crypto.timingSafeEqual. There's a good example of usage here: https://github.com/interledgerjs/ilp-plugin-btp/blob/024bf2044ec9f34a88b277725f24b8894dd47f49/src/index.ts#L803-L804

You basically replace the equality check by turning the token and the token you're checking into buffers. Then you compare your length, and then if they have the same length you call `crypto.timingSafeEqual`.